packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,113 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from 'common-common/src/cacheDecorator';
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  const cacheFunc = (ttl: number) =>
+    cacheDecorator.cacheMiddleware(ttl, cosmosRPCKeyGenerator);
+
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802'];
+  const completedCodes = [
+    '0803',
+    '0804',
+    '0805',
+    '0803220a0a080000000000000087',
+    '0803220a0a080000000000000100',
+    '0803220a0a080000000000000189',
+  ];
+
+  return async function cacheIfCacheworthyRPC(req, res, next) {
+    let body = req.body;
+    try {
+      body = JSON.parse(req.body);
+    } catch (err) {
+      console.log('Error parsing body: ', err);
+    }
+
+    const params = JSON.stringify(body?.params);
+    const method = body?.method;
+
+    if (/^(status|block|tx)/.test(method)) {
+      // LIVENESS CHECK  or TX Response: do not cache
+      return next();
```

Comment:
`status` and `block` might make sense to cache for 6 seconds, since that is a typical cosmos block time. It's nice to have the performance boost here.

`tx` we should still avoid caching - they are user-specific